diff -ru boost/test/utils/runtime/cla/argv_traverser.ipp boost.new/test/utils/runtime/cla/argv_traverser.ipp
--- boost/test/utils/runtime/cla/argv_traverser.ipp	2008-10-13 09:20:26.000000000 +0100
+++ boost.new/test/utils/runtime/cla/argv_traverser.ipp	2014-06-18 10:59:46.436275271 +0100
@@ -40,7 +40,7 @@
 
 BOOST_RT_PARAM_INLINE
 argv_traverser::argv_traverser()
-: p_ignore_mismatch( false ), p_separator( BOOST_RT_PARAM_LITERAL( ' ' ) )
+: noncopyable(), p_ignore_mismatch( false ), p_separator( BOOST_RT_PARAM_LITERAL( ' ' ) )
 {
 }
 
diff -ru boost/test/utils/trivial_singleton.hpp boost.new/test/utils/trivial_singleton.hpp
--- boost/test/utils/trivial_singleton.hpp	2008-10-13 09:20:26.000000000 +0100
+++ boost.new/test/utils/trivial_singleton.hpp	2014-06-18 10:59:06.799974274 +0100
@@ -37,7 +37,7 @@
 public:
     static Derived& instance() { static Derived the_inst; return the_inst; }    
 protected:
-    singleton()  {}
+    singleton() : noncopyable() {}
     ~singleton() {}
 };
 
