--- CMakeLists.txt	2012-08-17 18:23:44.000000000 +0100
+++ CMakeLists.txt.without-openmp	2012-08-17 18:24:03.000000000 +0100
@@ -28,15 +28,6 @@
 ############################################################################
 ### compiler flags
 
-# OpenMP
-find_package(OpenMP REQUIRED)
-if(NOT OPENMP_FOUND)
-   message(WARNING "Could not find working OpenMP. Please set manually.")
-else()
-   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-endif()
-
 if(CMAKE_BUILD_TYPE STREQUAL "Release")
    if(NOT ENABLE_64BIT)
       execute_process(COMMAND ${CMAKE_SOURCE_DIR}/cmake/compilerflags
