--- ./BoostTesting.cmake	2012-03-31 20:18:22.000000000 +0100
+++ ./BoostTesting.cmake.new	2012-05-06 13:56:56.000000000 +0100
@@ -49,7 +49,7 @@
 
 if (BUILD_TESTING)
   if (NOT EXISTS ${CMAKE_BINARY_DIR}/CTestCustom.cmake)
-    configure_file(${CMAKE_SOURCE_DIR}/tools/build/CMake/CTestCustom.cmake.in
+	  configure_file(${PROJECT_SOURCE_DIR}/tools/build/CMake/CTestCustom.cmake.in
       ${CMAKE_BINARY_DIR}/CTestCustom.cmake
       COPYONLY)
   endif()
