--- ./foamgridentity.hh	2011-09-13 15:39:25.000000000 +0200
+++ ./foamgridentity_new.hh	2011-11-20 22:18:58.000000000 +0100
@@ -146,7 +146,17 @@
     
     const FoamGridEntityImp<dim-codim,GridImp::dimensionworld>* target_;
 
-        
+
+	FoamGridEntity& operator=(const FoamGridEntity& original)
+	    {
+		if (this != &original)
+		{
+			geo_.reset();
+			target_ = original.target_;
+		}
+		return *this;
+	    }
+
     private:
     
         //! \todo Please doc me !
