--- include/tbb/pipeline.h	2013-10-16 13:44:31.000000000 +0100
+++ include/tbb/pipeline.h.new	2013-11-19 21:39:28.831045318 +0000
@@ -342,10 +342,11 @@ template<> struct tbb_trivially_copyable
 template<> struct tbb_trivially_copyable <float> { enum { value = !tbb_large_object<float>::value }; };
 template<> struct tbb_trivially_copyable <double> { enum { value = !tbb_large_object<double>::value }; };
 #else
-#if __GNUC__==4 && __GNUC_MINOR__>=4 && __GXX_EXPERIMENTAL_CXX0X__
-template<typename T> struct tbb_trivially_copyable { enum { value = std::has_trivial_copy_constructor<T>::value }; };
-#else
+#include <ciso646> // dummy header to see if _LIBCPP_VERSION symbol gets defined
+#ifdef _LIBCPP_VERSION
 template<typename T> struct tbb_trivially_copyable { enum { value = std::is_trivially_copyable<T>::value }; };
+#else
+template<typename T> struct tbb_trivially_copyable { enum { value = std::has_trivial_copy_constructor<T>::value }; };
 #endif //
 #endif // TBB_IMPLEMENT_CPP0X
 
