--- ACA.h.debug-messages	2013-02-13 16:53:01.000000000 +0000
+++ ACA.h.ACAn-inext-jnext	2013-02-13 16:54:14.000000000 +0000
@@ -385,7 +385,7 @@
           const cluster* c1, const cluster* c2)
 {
   typedef typename num_traits<T>::abs_type abs_T;
-  unsigned l, no = 0, j0, inext;
+  unsigned l, no = 0, j0, inext, jnext;
   // klast is the number of successful steps applied to the last pivotal row
   unsigned kmin, keps, k_old;
   abs_T nrmlsk2, nrms2 = 0.0;
@@ -422,10 +422,10 @@
 
     k_old = k;
 
+    inext = i0;
     do {
 
       // compute a cross
-      inext = i0;
 
       unsigned klast;
       ok = ACA_row_step(MatGen, b1, n1, b2, n2, klast, inext, k, no, Z, S,
@@ -501,10 +501,10 @@
 
     k_old = k;
 
+    jnext = j0;
     do {
 
       // compute a cross
-      unsigned jnext = j0;
       k_old = k;
       ok = ACA_col_step(MatGen, b1, n1, b2, n2, jnext, k, no, Z, S,
 			U, V, nrmlsk2, scale, c1, c2);
