--- Include/ACA.h.frobenius-norm	2013-02-13 16:50:31.000000000 +0000
+++ Include/ACA.h	2013-02-13 16:48:46.000000000 +0000
@@ -97,7 +97,8 @@
       const T e = (pv[l] *= sca);
       nrmv2 += abs2(e);
       if (abs(e) > 1e-8*scale*sqrtpiv) ++S[l];  // col l will be succ. approx
-    }
+    } else
+      pv[l] = 0.;
   }
 
   MatGen.cmpbl(b1, n1, b2+j0, 1, pu, c1, c2);
@@ -179,7 +183,8 @@
       const T e = (pu[l] *= sca);
       nrmu2 += abs2(e);
       if (abs(e) > 1e-8 * scale*sqrtpiv) ++Z[l];
-    }
+    } else
+      pu[l] = 0.;
   }
 
   MatGen.cmpbl(b1+i0, 1, b2, n2, pv, c1, c2);
