--- Include/ACA.h.w-matgen-patch	2013-02-13 15:44:13.000000000 +0000
+++ Include/ACA.h	2013-02-13 16:50:31.000000000 +0000
@@ -263,7 +263,7 @@
 
     sum = (T) 0.0;                            // update nrms2
     for (l=0; l<k; ++l)
-      sum += blas::scpr(n1, U+l*n1, U+k*n1) * blas::scpr(n2, V+l*n2, V+k*n2);
+      sum += blas::scpr(n1, U+l*n1, U+k*n1) * blas::scpr(n2, V+k*n2, V+l*n2);
 
     nrms2 += 2.0 * Re(sum) + nrmlsk2;
 
@@ -415,7 +415,7 @@
       // else: step was successful, check stopping criterion      
       sum = (T) 0.0;                            // update nrms2
       for (l=0; l<k; ++l)
-	sum += blas::scpr(n1, U+l*n1, U+k*n1) * blas::scpr(n2, V+l*n2, V+k*n2);
+        sum += blas::scpr(n1, U+l*n1, U+k*n1) * blas::scpr(n2, V+k*n2, V+l*n2);
 
       nrms2 += 2.0 * Re(sum) + nrmlsk2;
 
@@ -494,7 +494,7 @@
       
       sum = (T) 0.0;                            // update nrms2
       for (l=0; l<k; ++l)
-	sum += blas::scpr(n1, U+l*n1, U+k*n1) * blas::scpr(n2, V+l*n2, V+k*n2);
+        sum += blas::scpr(n1, U+l*n1, U+k*n1) * blas::scpr(n2, V+k*n2, V+l*n2);
 
       nrms2 += 2.0 * Re(sum) + nrmlsk2;
 
@@ -701,7 +701,7 @@
 
     sum = (T) 0.0;                            // update nrms2
     for (l=0; l<k; ++l)
-      sum += blas::scpr(n1, U+l*n1, U+k*n1) * blas::scpr(n2, V+l*n2, V+k*n2);
+      sum += blas::scpr(n1, U+l*n1, U+k*n1) * blas::scpr(n2, V+k*n2, V+l*n2);
 
     nrms2 += 2.0 * Re(sum) + nrmlsk2;
 
