--- Include/bemcluster.h.orig	2013-05-01 15:02:28.000000000 +0100
+++ Include/bemcluster.h	2013-04-29 18:55:08.000000000 +0100
@@ -8,6 +8,14 @@
     this software; if not, see AHMED's internet site.
 */
 
+/*
+    This file has been modified by the authors of BEM++.
+
+    Summary of changes:
+
+    - 13/02/2013: corrected a bug in the assignment of icom during construction
+      of bemcluster.
+*/
 
 #ifndef BEMCLUSTER_H
 #define BEMCLUSTER_H
