--- apprx.h.debug-messages	2013-02-13 17:13:05.000000000 +0000
+++ apprx.h.total_pivot_count	2013-02-13 17:11:53.000000000 +0000
@@ -20,6 +20,10 @@
 #include "bemblcluster.h"
 #include "ACA.h"
 
+#include <tbb/atomic.h>
+
+tbb::atomic<size_t> total_pivot_count;
+
 #ifdef CHECK_ACA_ERROR
 
 double ACA_error_max = 0.0;
@@ -169,6 +173,7 @@
 
     succ = aca_routine(MatGen, b1, n1, b2, n2, eps, maxk, i0, k, U, V,
                        bl->getcl1(), bl->getcl2());
+    total_pivot_count += k;
 
     if (succ) {
       mbl->cpyLrM_cmpr(k, U, n1, V, n2, eps, k);
