--- Include/ACA.h.orig	2013-04-28 15:02:28.000000000 +0100
+++ Include/ACA.h	2013-04-29 18:55:08.000000000 +0100
@@ -8,6 +8,21 @@
     this software; if not, see AHMED's internet site.
 */
 
+/*
+    This file has been modified by the authors of BEM++.
+
+    Summary of changes:
+
+    - 17/11/2012: added parameters c1 and c2 to routines ACA_row_step() and
+      ACA_col_step().
+    - 13/02/2013: corrected the formula used to update the Frobenius norm
+      (lines "sum += blas::scpr...").
+    - 13/02/2013: in routines ACA_row_step() and ACA_col_step(), rows/columns
+      (respectively) used as pivots in previous steps are zeroed.
+    - 09/03/2013: added arguments retry_if_zero, orig_row and orig_col to
+      routines ACA_row_step() and ACA_col_step() and changed return type to
+      ACA_status.
+*/
 
 #ifndef ACA_H
 #define ACA_H
