--- Include/apprx.h.orig	2012-06-11 15:50:28.000000000 +0100
+++ Include/apprx.h	2012-11-17 14:54:20.000000000 +0000
@@ -34,7 +34,7 @@
   mbl->convLrM_toGeM(A, n1);
   T *B = new T[n1*n2];
 
-  MatGen.cmpbl(bl->getb1(), n1, bl->getb2(), n2, B);
+  MatGen.cmpbl(bl->getb1(), n1, bl->getb2(), n2, B, bl->getcl1(), bl->getcl2());
 
   double dnrm = 0.0, dmax = 0.0;
   for (unsigned j=0; j<n2; ++j) {
@@ -106,7 +106,8 @@
 
     unsigned i0 = bl->getcl1()->geticom() - b1;
 
-    succ = ::ACAr(MatGen, b1, n1, b2, n2, eps, maxk, i0, k, U, V);
+    succ = ::ACAr(MatGen, b1, n1, b2, n2, eps, maxk, i0, k, U, V,
+                  bl->getcl1(), bl->getcl2());
 
     if (succ) {
       mbl->cpyLrM_cmpr(k, U, n1, V, n2, eps, k);
@@ -123,10 +124,10 @@
       if(cmplx_sym) mbl->setSyM();
       else mbl->setHeM();
       
-      MatGen.cmpblsym(b1, n1, mbl->getdata());
+      MatGen.cmpblsym(b1, n1, mbl->getdata(), bl->getcl1());
     } else {
       mbl->setGeM();
-      MatGen.cmpbl(b1, n1, b2, n2, mbl->getdata());
+      MatGen.cmpbl(b1, n1, b2, n2, mbl->getdata(), bl->getcl1(), bl->getcl2());
     }
   }
 
@@ -153,7 +154,8 @@
     T *U, *V;
 
     unsigned i0 = bl->getcl1()->geticom() - b1;
-    succ = ::ACA(MatGen, b1, n1, b2, n2, eps, maxk, i0, k, U, V);
+    succ = ::ACA(MatGen, b1, n1, b2, n2, eps, maxk, i0, k, U, V,
+                 bl->getcl1(), bl->getcl2());
 
     if (succ) {
       mbl->cpyLrM_cmpr(k, U, n1, V, n2, eps, k);
@@ -167,7 +169,7 @@
 
   if (!succ) {
     mbl->setGeM();
-    MatGen.cmpbl(b1, n1, b2, n2, mbl->getdata());
+    MatGen.cmpbl(b1, n1, b2, n2, mbl->getdata(), bl->getcl1(), bl->getcl2());
   }
 }
 