--- bemcluster.h	2013-02-13 17:16:37.000000000 +0000
+++ bemcluster.h.permuted	2013-02-13 17:16:27.000000000 +0000
@@ -25,25 +25,26 @@
   bemcluster(T* dofs, unsigned* op_perm, unsigned k, unsigned l) : cluster3d_pca<T>(dofs, op_perm, k, l)
   {
     // compute icom
-    icom = this -> nbeg;
+    icom = op_perm[this->nbeg]; // original index. Will be replaced by permuted
+                                // index in createClusterTree()
 
     const unsigned n = cluster3d_pca<T>::dim();
     unsigned i, j;
 
     double smin = 0.0;
     for (j=0; j<n; ++j) {
-      const double e = cluster3d_pca<T>::getcom(j) - dofs[this->nbeg].getcenter(j);
+      const double e = cluster3d_pca<T>::getcom(j) - dofs[op_perm[this->nbeg]].getcenter(j);
       smin += e*e;
     }
 
     for (i=this->nbeg+1; i<this->nend; ++i) {
       double s = 0.0;
       for (j=0; j<n; ++j) {
-        const double e = cluster3d_pca<T>::getcom(j) - dofs[i].getcenter(j);
+        const double e = cluster3d_pca<T>::getcom(j) - dofs[op_perm[i]].getcenter(j);
         s += e*e;
       }
       if (s<smin) {
-        icom = i;
+        icom = op_perm[i];
         smin = s;
       }
     }
@@ -56,6 +57,15 @@
 
   //! returns the icom index
   unsigned geticom() const { return icom; }
+
+  // once all descendants are created (so that the array po_perm[nbeg...nend)
+  // won't change any more), set icom to the permuted index of the centroid
+  virtual void createClusterTree(const unsigned bmin,
+                                 unsigned* op_perm, unsigned* po_perm)
+  {
+    cluster3d_pca<T>::createClusterTree(bmin, op_perm, po_perm);
+    icom = po_perm[icom];
+  }
 };
 
 
