--- ./structuredgridfactory.hh	2011-01-24 13:33:37.000000000 +0000
+++ ./structuredgridfactory_new.hh	2011-11-17 23:17:54.000000000 +0000
@@ -81,8 +81,8 @@
         
         /** \brief Insert a structured set of vertices into the factory */
         static void insertVertices(GridFactory<GridType>& factory,
-                                   const FieldVector<ctype,dim>& lowerLeft,
-                                   const FieldVector<ctype,dim>& upperRight,
+                                   const FieldVector<ctype,dimworld>& lowerLeft,
+                                   const FieldVector<ctype,dimworld>& upperRight,
                                    const array<unsigned int,dim>& vertices)
         {
 
