--- dune-grid/dune/grid/yaspgrid/grids.hh	2013-01-19 13:31:56.000000000 +0000
+++ dune-grid/dune/grid/yaspgrid/grids.hh.new	2013-11-18 11:26:45.000000000 +0000
@@ -1246,7 +1246,7 @@ namespace Dune {
     int rank_relative (int rank, int dir, int cnt) const
     {
       iTupel coord = rank_to_coord(rank);
-      coord[dir] = (coord[dir]+dims[dir]+cnt)%dims[dir];
+      coord[dir] = (coord[dir]+dims(dir)+cnt)%dims(dir);
       return coord_to_rank(coord);
     }
 
