--- ./packages/epetra/src/Epetra_ConfigDefs.h
+++ ./packages/epetra/src/Epetra_ConfigDefs.h
@@ -163,7 +163,7 @@ For xlC 12.1, malloc and related functions are provided outside the std
 namespace, so the below three using statements cause conflicting declarations.
 */
 #ifndef __clang__
-#if !defined __IBMCPP__ || ( __IBMCPP__ != 1210 )
+#if !defined __IBMCPP__ || ( (__IBMCPP__+0) != 1210 )
 using std::realloc;
 using std::malloc;
 using std::free;
