--- ./packages/stratimikos/adapters/belos/src/Thyra_BelosLinearOpWithSolve_def.hpp	2012-04-24 20:30:54.000000000 +0100
+++ ./packages/stratimikos/adapters/belos/src/Thyra_BelosLinearOpWithSolve_def_new.hpp	2012-05-03 17:34:53.000000000 +0100
@@ -124,6 +124,7 @@
   const int convergenceTestFrequency
   )
 {
+  typedef typename Teuchos::ScalarTraits<Scalar>::magnitudeType MagnitudeType;
   this->setLinePrefix("BELOS/T");
   // ToDo: Validate input
   lp_ = lp;
@@ -139,13 +140,13 @@
   // not, use the default from the solver.
   if ( !is_null(solverPL_) ) {
     if (solverPL_->isParameter("Convergence Tolerance")) {
-      defaultTol_ = solverPL_->get<double>("Convergence Tolerance");
+      defaultTol_ = solverPL_->get<MagnitudeType>("Convergence Tolerance");
     }
   }
   else {
     RCP<const Teuchos::ParameterList> defaultPL =
       iterativeSolver->getValidParameters();
-    defaultTol_ = defaultPL->get<double>("Convergence Tolerance");
+    defaultTol_ = defaultPL->get<MagnitudeType>("Convergence Tolerance");
   }
 }
 
