--- ./packages/stratimikos/adapters/belos/src/Thyra_BelosLinearOpWithSolve_def.hpp	2014-01-29 23:00:36.000000000 +0000
+++ ./packages/stratimikos/adapters/belos/src/Thyra_BelosLinearOpWithSolve_def.hpp.patched	2014-02-06 15:03:42.000000000 +0000
@@ -167,6 +167,7 @@
   const int convergenceTestFrequency
   )
 {
+  typedef typename Teuchos::ScalarTraits<Scalar>::magnitudeType MagnitudeType;
   this->setLinePrefix("BELOS/T");
   // ToDo: Validate input
   lp_ = lp;
@@ -182,13 +183,13 @@
   // not, use the default from the solver.
   if ( !is_null(solverPL_) ) {
     if (solverPL_->isParameter("Convergence Tolerance")) {
-      defaultTol_ = solverPL_->get<double>("Convergence Tolerance");
+      defaultTol_ = solverPL_->get<MagnitudeType>("Convergence Tolerance");
     }
   }
   else {
     RCP<const Teuchos::ParameterList> defaultPL =
       iterativeSolver->getValidParameters();
-    defaultTol_ = defaultPL->get<double>("Convergence Tolerance");
+    defaultTol_ = defaultPL->get<MagnitudeType>("Convergence Tolerance");
   }
 }
 
