--- foamgrid.hh.old	2011-11-22 13:19:05.000000000 +0000
+++ foamgrid.hh	2011-11-22 15:46:03.000000000 +0000
@@ -408,13 +408,11 @@
         {}
 #endif
         
-#if 0        
         /** dummy collective communication */
-        const CollectiveCommunication& comm () const
+        const typename Traits::CollectiveCommunication& comm () const
         {
             return ccobj_;
         }
-#endif   
         /*@}*/
         
         
@@ -446,10 +444,10 @@
         // IdSets don't need updating
 
         }
-#if 0         
-        //! \todo Please doc me !
-        CollectiveCommunication ccobj_;
-#endif
+
+        //! Collective communication interface
+        typename Traits::CollectiveCommunication ccobj_;
+
     // Stores the lists of vertices, edges, elements for each level
     std::vector<tuple<std::list<FoamGridEntityImp<0,dimworld> >,
                       std::list<FoamGridEntityImp<1,dimworld> >,
