--- include/armadillo_bits/config.hpp.cmake	2014-06-01 03:44:40.000000000 +0100
+++ include/armadillo_bits/config.hpp.cmake.new	2014-07-04 15:21:22.000000000 +0100
@@ -83,12 +83,12 @@
 //// Note that ARMA_USE_U64S64 is automatically enabled when ARMA_64BIT_WORD or ARMA_USE_CXX11 are enabled
 #endif
 
-#if !defined(ARMA_USE_HDF5)
-#cmakedefine ARMA_USE_HDF5
+//// #if !defined(ARMA_USE_HDF5)
+//// #cmakedefine ARMA_USE_HDF5
 //// Uncomment the above line to allow the ability to save and load matrices stored in HDF5 format;
 //// the hdf5.h header file must be available on your system,
 //// and you will need to link with the hdf5 library (eg. -lhdf5)
-#endif
+//// #endif
 
 #if !defined(ARMA_MAT_PREALLOC)
   #define ARMA_MAT_PREALLOC 16
