--- dune-common/cmake/modules/DuneMacros.cmake
+++ dune-common/cmake/modules/DuneMacros.cmake.new
@@ -551,7 +551,9 @@ macro(dune_project)
 
   # optional Fortran support
   include(LanguageSupport)
-  workaround_9220(Fortran Fortran_Works)
+  if(ENABLE_Fortran)
+    workaround_9220(Fortran Fortran_Works)
+  endif()
   if(Fortran_Works)
     enable_language(Fortran OPTIONAL)
   endif(Fortran_Works)
