--- dune-grid/dune/grid/io/file/dgfparser/blocks/basic.hh	2013-01-19 11:23:52.000000000 +0000
+++ dune-grid/dune/grid/io/file/dgfparser/blocks/basic.hh.new	2013-11-17 21:51:57.000000000 +0000
@@ -58,7 +58,7 @@ namespace Dune
       bool getnextentry( ENTRY &entry )
       {
         line >> entry;
-        return line;
+        return (bool) line;
       }
 
       bool gettokenparam ( std :: string token, std :: string &entry );
